MdeModulePkg/Include/Protocol/MediaSanitize.h

```diff
@@ -0,0 +1,173 @@
+/** @file
+  This file defines the Media Sanitize Protocol.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#ifndef __MEDIA_SANITIZE_PROTOCOL_H__
+#define __MEDIA_SANITIZE_PROTOCOL_H__
+
+#define MEDIA_SANITIZE_PROTOCOL_GUID \
+  { \
+    0x0d799a99, 0x25af, 0x429e, { 0x92, 0x72, 0xd0, 0xb2, 0x7d, 0x6d, 0x5f, 0x14 } \
+  }
+
+typedef struct _MEDIA_SANITIZE_PROTOCOL MEDIA_SANITIZE_PROTOCOL;
+
+#define MEDIA_SANITIZE_PROTOCOL_REVISION  0x00010000
+
+///
+/// Sanitize actions for purge operation.
+///
+/// NOTE: First four actions (no action, overwrite, block erase, crypto erase) cannot
+/// be overlapped. All other fields may be overlapped as they apply.
+///
+#define PURGE_ACTION_NO_ACTION                         0x00000000 // No purge action requested
+#define PURGE_ACTION_OVERWRITE                         0x00000001 // Overwrite with 32-bit pattern
+#define PURGE_ACTION_BLOCK_ERASE                       0x00000002 // Erase Blocks with indeterminate pattern
+#define PURGE_ACTION_CRYPTO_ERASE                      0x00000004 // Delete encryption keys only
+#define PURGE_ACTION_RESET_REQUIRED                    0x00000008 // Reset required after purge
+#define PURGE_ACTION_NO_DEALLOCATE                     0x00000010 // Do no deallocate (trim) flash medai after sanitize

```

Comment:
nit: should it be below?
```suggestion
#define PURGE_ACTION_NO_DEALLOCATE                     0x00000010 // Do not deallocate (trim) flash medai after sanitize

```

Reply:
also fix spelling for media